extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/CommonCacheNotifier.java

```diff
@@ -106,23 +126,26 @@ public void start()
               Pair<String, byte[]> update = updateQueue.take();
               String authorizer = update.lhs;
               byte[] serializedMap = update.rhs;
-              BasicAuthDBConfig authorizerConfig = itemConfigMap.get(update.lhs);
-              if (!authorizerConfig.isEnableCacheNotifications()) {
+
+              BasicAuthDBConfig dbConfig = itemConfigMap == null ? itemConfig : itemConfigMap.get(update.lhs);
+              if (!dbConfig.isEnableCacheNotifications()) {
                 continue;
               }
 
               LOG.debug(callerName + ":Sending cache update notifications");
               // Best effort, if a notification fails, the remote node will eventually poll to update its state
               // We wait for responses however, to avoid flooding remote nodes with notifications.
-              List<ListenableFuture<StatusResponseHolder>> futures = sendUpdate(
-                  authorizer,
-                  serializedMap
-              );
+              List<ListenableFuture<StatusResponseHolder>> futures;
+              if (authorizer != null) {
```

Comment:
The authorizer will always be non-null here, the `addUpdate` method that would have allowed it to be null is not used

Reply:
Done, updated